Terminal.Gui/View/View.cs

```diff
@@ -2657,16 +2650,27 @@ internal static List<View> TopologicalSort (View superView, IEnumerable<View> no
 			}
 
 			if (edges.Any ()) {
-				(var from, var to) = edges.First ();
-				if (from != superView?.GetTopSuperView (to, from)) {
-					if (!ReferenceEquals (from, to)) {
+				foreach ((var from, var to) in edges) {
+					if (from == to) {
+						// if not yet added to the result, add it and remove from edge
+						if (result.Find (v => v == from) == null) {
+							result.Add (from);
+						}
+						edges.Remove ((from, to));
+					} else if (from.SuperView == to.SuperView) {
+						if (result.Find (v => v == from) == null) {
+							result.Add (from);
+						}
+						if (result.Find (v => v == to) == null) {
+							result.Add (to);
+						}
+						edges.Remove ((from, to));
+					} else if (from != superView?.GetTopSuperView (to, from) && !ReferenceEquals (from, to)) {
 						if (ReferenceEquals (from.SuperView, to)) {
 							throw new InvalidOperationException ($"ComputedLayout for \"{superView}\": \"{to}\" references a SubView (\"{from}\").");
 						} else {
 							throw new InvalidOperationException ($"ComputedLayout for \"{superView}\": \"{from}\" linked with \"{to}\" was not found. Did you forget to add it to {superView}?");
 						}
-					} else {
```

Comment:
Is it not possible to detect a recursive cycle?

Or, is a recursive cycle impossible?

Or, is it not a bad thing to have a recursive cycle?

Reply:
I couldn't get a unfinished recursive cycle. I only get some recursive cycle when it is expected if a subview references a superview or a super-superview, but it ended this recursive cycle as expected. If we caught an unfinished recursive cycle in the future we'll handle it.